recipes/emscripten-activation/activate.sh

```diff
@@ -0,0 +1,62 @@
+if [ -z ${CONDA_FORGE_EMSCRIPTEN_ACTIVATED+x} ]; then
+
+    export CONDA_FORGE_EMSCRIPTEN_ACTIVATED=1
+
+    export EMSDK_PYTHON=${CONDA_PREFIX}/bin/python3
+    export PYTHON=${CONDA_PREFIX}/bin/python3
+
+    CONDA_EMSDK_DIR=$CONDA_PREFIX/opt/emsdk
+ 
+    export EMSCRIPTEN_VERSION=$PKG_VERSION
+    export EMSCRIPTEN_FORGE_EMSDK_DIR=$CONDA_EMSDK_DIR
+
+    # clear all prexisting cmake args / CC / CXX / AR / RANLIB
+    export CC="emcc"
+    export CXX="em++"
+    export AR="emar"
+    export RANLIB="emranlib"
+    
+    export CMAKE_ARGS=""
+
+    # set the emscripten toolchain
+    export CMAKE_ARGS="$CMAKE_ARGS -DCMAKE_TOOLCHAIN_FILE=$CONDA_EMSDK_DIR/upstream/emscripten/cmake/Modules/Platform/Emscripten.cmake"
+
+    # conda prefix path
+    export CMAKE_ARGS="$CMAKE_ARGS -DCMAKE_PREFIX_PATH=$PREFIX"
+
+    # install prefix
+    export CMAKE_ARGS="$CMAKE_ARGS -DCMAKE_INSTALL_PREFIX=$PREFIX"
+
+    # find root path mode package
+    export CMAKE_ARGS="$CMAKE_ARGS -DCMAKE_FIND_ROOT_PATH_MODE_PACKAGE=ON"
+
+    # fpic
+    export CMAKE_ARGS="$CMAKE_ARGS -DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=true"
+
+    cmake () {
+        emcmake cmake "$@"
+    }
```

Comment:
Why is this needed? If it's needed, then packages that use cmake indirectly will not work.

Reply:
Is that a realistic concern, given the target platform? AFAIU, we won't be able to build anything with vanilla cmake. I guess the main value add is to avoid having to do a bunch of renames.

Would symlinking be more acceptable?

Reply:
Yes, it is a realistic concern. Projects like scikit-build use cmake under the hood. We want to be able to compile recipes without too much divergence or patching to avoid the maintenance burden.

Reply:
Of course usage of `cmake` is extremely wide-spread, that was not what I meant. AFAIU, vanilla cmake isn't able to build for emscripten though, and we always(?) need to go through `emcmake` when building for that target.

> We want to be able to compile recipes without too much divergence or patching to avoid the maintenance burden.

I think that was actually the idea here - by avoiding the distinction between `cmake` and `emcmake` for every recipe, we'd reduce divergence. I didn't write this though, so perhaps @DerThorsten or @wolfv want to explain the reasoning.

Reply:
This is a bash function and will work only in the same bash session. If cmake is called from a python program called from bash, then vanilla cmake will be called.

> AFAIU, vanilla cmake isn't able to build for emscripten though, and we always(?) need to go through emcmake when building for that target.

Have a look at emcmake. All it does is send `CMAKE_TOOLCHAIN_FILE`, `CMAKE_CROSSCOMPILING_EMULATOR` and change the default generator in windows.

Reply:
Ok, thanks for the explanation 🙏

Reply:
Removed.